api/src/lib/cart/src/get-cart.js

```diff
@@ -0,0 +1,38 @@
+/**
+ * @param {cartPgRepo} cartRepo
+ */
+function GetcartFactory(deps) {
+    'use strict'
+
+    const { cartRepo } = deps
+
+    const getCart = async (ctx) => {
+        const logger = ctx.request.state.logger
+        var productids = []
+
+        try {
+
+            const cart = await cartRepo.get.byId(ctx.state.session.id)
+            for (var i = 0; i < cart.length; i++) {
+                productids.push(cart[i].productid)
+            }
```

Comment:
`var` is no longer commonly used, and the way that it's used here introduces scope confusion. That's because `var` is not block scoped, but as written, it _appears_ to be block scoped. `let` and `const` are both block scoped. Since `i` should be block scoped, and it gets mutated, `let` is the appropriate keyword to use instead of `var` in this code.

That being said, this approach is unnecessarily complex, and risks astonishing bugs during refactoring. You could use functional programming instead:

```javascript
const productIds = cart.map((item) => item.productid)
const products = await cartRepo.get.prod(productIds)
```